Detectors/EMCAL/base/src/ClusterFactory.cxx

```diff
@@ -551,24 +555,145 @@ void ClusterFactory<InputType>::evalElipsAxis(gsl::span<const int> inputsIndices
+    LOG(debug) << Form("EXOTIC CELL id %d, eCell %f, eCross %f, 1-eCross/eCell %f",
+                       towerId, ecell, eCross, 1 - eCross / ecell);
+    return true;
+  }
+
+  return false;
+}
+
+///
+///  Calculate the energy in the cross around the energy of a given cell.
+//____________________________________________________________________________
+template <class InputType>
+float ClusterFactory<InputType>::getECross(short towerId, float energy) const
+{
+  auto [iSM, iMod, iIphi, iIeta] = mGeomPtr->GetCellIndex(towerId);
+  auto [iphi, ieta] = mGeomPtr->GetCellPhiEtaIndexInSModule(iSM, iMod, iIphi, iIeta);
+
+  // Get close cells index, energy and time, not in corners
+
+  short towerId1 = -1;
+  short towerId2 = -1;
+
+  if (iphi < o2::emcal::EMCAL_ROWS - 1)
+    towerId1 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi + 1, ieta);
+  if (iphi > 0)
+    towerId2 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi - 1, ieta);
+
+  // In case of cell in eta = 0 border, depending on SM shift the cross cell index
+
+  short towerId3 = -1;
+  short towerId4 = -1;
+
+  if (ieta == o2::emcal::EMCAL_COLS - 1 && !(iSM % 2)) {
+    towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM + 1, iphi, 0);
+    towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta - 1);
+  } else if (ieta == 0 && iSM % 2) {
+    towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
+    towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM - 1, iphi, o2::emcal::EMCAL_COLS - 1);
+  } else {
+    if (ieta < o2::emcal::EMCAL_COLS - 1)
+      towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
+    if (ieta > 0)
+      towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta - 1);
+  }
+
+  LOG(debug) << Form("iSM %d, towerId %d, a %d, b %d, c %d e %d", iSM, towerId, towerId1, towerId2, towerId3, towerId4);
+
+  float ecell1 = 0, ecell2 = 0, ecell3 = 0, ecell4 = 0;
+
+  short iTowerId = -1;
+  for (auto& cellIndex : mCellsIndices) {
+    iTowerId = mInputsContainer[cellIndex].getTower();
+    if (towerId1 == iTowerId) {
+      ecell1 = mInputsContainer[cellIndex].getEnergy();
+    }
+    if (towerId2 == iTowerId) {
+      ecell1 = mInputsContainer[cellIndex].getEnergy();
+    }
+    if (towerId3 == iTowerId) {
+      ecell1 = mInputsContainer[cellIndex].getEnergy();
+    }
+    if (towerId4 == iTowerId) {
+      ecell1 = mInputsContainer[cellIndex].getEnergy();
+    }
+  }
+
+  float w1 = 1, w2 = 1, w3 = 1, w4 = 1;
+  if (mUseWeightExotic) {
+    w1 = GetCellWeight(ecell1, energy);
+    w2 = GetCellWeight(ecell2, energy);
+    w3 = GetCellWeight(ecell3, energy);
+    w4 = GetCellWeight(ecell4, energy);
+  }
+
+  if (ecell1 < mExoticCellInCrossMinAmplitude || w1 <= 0)
+    ecell1 = 0;
+  if (ecell2 < mExoticCellInCrossMinAmplitude || w2 <= 0)
+    ecell2 = 0;
+  if (ecell3 < mExoticCellInCrossMinAmplitude || w3 <= 0)
+    ecell3 = 0;
+  if (ecell4 < mExoticCellInCrossMinAmplitude || w4 <= 0)
+    ecell4 = 0;
```

Comment:
The original implementation has also a condition that the time difference is within a certain range.